org.civicrm.reqangular/src/common/angular-date/src/services/DateValidationService.js

```diff
@@ -56,7 +56,7 @@ function DateValidationService($filter) {
 
     me.validate = function validate(start_date, end_date) {
```

Comment:
I haven't seen this file before, so now I can't comment on it except for this few lines that you changed from the last PR. I'm gonna bundle every note here then

----
```js
me.start_parts = [];
me.end_parts = [];
```
*(wherever else applies)*
You are mixing `snake_case` and `camelCase` for variable names, please stick to `camelCase`

---
```js
me._error = function (error_msg, fields) {
    throw new Error(error_msg);
};
```
*(wherever else applies)*
No matter if we use CommonJS or AMD, if a function is private (I believe that's why you prefixed it with `_`), you can just put it outside of the object/function you are returning from the module. This way the module api is comprised of actual "public methods" only:
```js
function error (error_msg, fields) {
    throw new Error(error_msg);
};

function DateValidationService($filter) {
    // ...
}

module.exports = DateValidationService;
```

---
```js
me._error = function (error_msg, fields) {
    throw new Error(error_msg);
};
```
It doesn't seem you are using `fields` for anything?

---

```js
if (!!(error && error.constructor && error.call && error.apply)) {
    me._error = error;
} else {
    throw new TypeError('Error callback must be a function.');
}
```
Why not simply `typeof error === 'function'`, or since we have [Lodash](https://lodash.com/) as a common dependency, you could inject that and use the [`_.isFunction`](https://lodash.com/docs#isFunction) utility

---

```js
if(!!start_date)
```
*(wherever else applies)*
JS already coerces the value to a boolean inside an `if()`, it's not necessary to do it explicitly

---
```js
if(!me.start_date || !!start_date) me.setDates(start_date, end_date);
```
*(wherever else applies)*
Please don't use this inline form, for consistency rewrite as:
```js
if (!me.start_date || !!start_date) {
    me.setDates(start_date, end_date);
}
```
----
```js
if(me.end) {
```
*(wherever else applies)*
Make sure the code style is consistent, put a space between `if` and `(` like everywhere else
```js
if (me.end) {
```

----
```js
me.setOptions = function setOptions(newOptions) {
    angular.extend(me.options, newOptions);
};
```
You are using `angular` as a global object instead of injecting it into the module

---
You think it would be better to use [Moment.js](http://momentjs.com/) in this service? It has already built in functionality for parsing dates, checking if a date is before/after an another, checking if the date is valid and so on